src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -1019,28 +1013,37 @@ var ReactDOMFiberComponent = {
           if (expectedStyle !== serverValue) {
             warnForPropDifference(propKey, serverValue, expectedStyle);
           }
-        } else if (
-          isCustomComponentTag ||
-          DOMProperty.isCustomAttribute(propKey)
-        ) {
+        } else if (isCustomComponentTag) {
           // $FlowFixMe - Should be inferred as not undefined.
-          extraAttributeNames.delete(propKey);
+          extraAttributeNames.delete(propKey.toLowerCase());
```

Comment:
Interestingly, it looks there's a related todo a earlier in the document:
https://github.com/facebook/react/blob/master/src/renderers/dom/fiber/ReactDOMFiberComponent.js#L931

Accessing the attribute name from the DOM is always going to return a lower cased name, so we need to do this for regular DOM elements too. For consistency, and until we close the thread on this, I've added an adjustment and test in 83e46aa.

I think we really just need to decide if we care about allowing custom attributes to have custom casing. If not, we should add that logic to `shouldSetAttribute` and probably make a separate warning along the lines of "Custom attributes must be lowercase".